### PR TITLE
more Doxygen updates to documentation

### PR DESCRIPTION
**Pull request by jbathegit**
Part of #48

### CODE REVIEW

File: src/cwbmg.c [hunk 1]
```diff
@@ -13,11 +13,10 @@
...
  *  @date 2005-11-29
  *
- *  @param[in] bmg     - character*(*): array containing BUFR
- *                        message to be written
- *  @param[in] nmb     - integer: size (in bytes) of BUFR message
+ *  @param[in] bmg     - char*: BUFR message to be written
+ *  @param[in] nmb     - f77int: Size (in bytes) of BUFR message
  *                        in bmg
- *  @param[out] iret    - integer: return code
+ *  @param[out] iret    - f77int: return code
```
Comment (aerorahul) [inline src/cwbmg.c:19]: what is f77int?

Comment (jbathegit) [reply src/cwbmg.c:19]: It's a typedef defined in bufrlib.h, equivalent to either an int or a long depending on whether it's a _4 or _8 build.

Comment (aerorahul) [reply src/cwbmg.c:19]: Thanks!

**Review by aerorahul — APPROVED** [review]
lots of dox updates.

**Comment by edwardhartnett** [comment]
Excellent work cleaning up these documentation blocks! Do we have one of our github workflows building the docs?

If not, you might want to add that. Simply make sure that doxygen is apt-get installed, and then add -DENABLE_DOCS=Yes to the cmake command in one of the workflows in .github/workflows...

**Pull request merged**